server/core/src/irods_signal.cpp

```diff
@@ -1,38 +1,84 @@
 extern "C"
+void stacktrace_signal_handler(int _signal)
 {
-    /// @brief Signal handler for seg faults
-    static void segv_handler(int _signal)
-    {
-        using log = irods::experimental::log;
+    timespec ts;
+
+    if (clock_gettime(CLOCK_REALTIME, &ts) != 0) {
+        std::exit(_signal);
+    }
 
-        std::stringstream ss;
-        ss << "Caught signal [" << _signal << "]. Dumping stacktrace and exiting ...\n";
+    const auto max_path_length = 512;
 
-        log::server::error({{"log_message", ss.str()},
-                            {"stacktrace", irods::stacktrace().dump()}});
+    char file_path_hidden[max_path_length] = "../../";
+    std::strcat(file_path_hidden, IRODS_DEFAULT_PATH_HOMEDIR);
+    std::strcat(file_path_hidden, "/stacktrace/");
 
-        exit(_signal);
-    }
+    // At this point, "file_path_hidden" should have the following path:
+    //
+    //     ../../<install_prefix>/stacktrace/
+    //
+
+    auto* buf_start = file_path_hidden + strlen(file_path_hidden);
+    auto* buf_end = file_path_hidden + sizeof(file_path_hidden);
+
+    if (auto [sec_end, ec] = std::to_chars(buf_start, buf_end, ts.tv_sec); std::errc{} == ec) {
+        *sec_end = '.';
+
+        if (auto [msec_end, ec] = std::to_chars(sec_end + 1, buf_end, ts.tv_nsec / 1'000'000);
+            std::errc{} == ec)
+        {
+            *msec_end = '.';
 
-    void register_handlers()
-    {
-        struct sigaction action;
-        memset(&action, 0, sizeof(action));
-        action.sa_handler = segv_handler;
-        sigaction(SIGSEGV, &action, 0);
-        sigaction(SIGABRT, &action, 0);
-        sigaction(SIGINT, &action, 0);
+            if (auto [pid_end, ec] = std::to_chars(msec_end + 1, buf_end, getpid()); std::errc{} == ec) {
+                // Terminate the hidden file path string. The filename should have the following format:
+                //
+                //     <epoch_seconds>.<remaining_milliseconds>.<agent_pid>
+                //
+                // This keeps the files sorted by timestamp and by pid.
+                *pid_end = 0;
+
+                char file_path[max_path_length];
+                std::strcpy(file_path, file_path_hidden);
+                
+                // An indicator to the thread handling the logging of these files that this file
+                // is not ready to be logged (i.e. the signal handler is still writing to it).
+                std::strcat(file_path_hidden, ".not_ready_for_logging");
+
+                // WRITE THE STACKTRACE!
+                boost::stacktrace::safe_dump_to(file_path_hidden);
+
+                // Now the stacktrace has been dumped, rename the file so that the trailing
+                // suffix (i.e. ".not_ready_for_logging") is removed.
+                std::rename(file_path_hidden, file_path);
+            }
+        }
     }
-}
+
+    std::exit(_signal);
+} // stacktrace_signal_handler
+
+void register_handlers() // TODO Consider calling this for grandpa also.
```

Comment:
add the word `signal` to this name.

suggestions:
- `set_signal_handlers`
- `register_signal_handlers`
- `set_unrecoverable_signal_handlers`
- `configure_unrecoverable_signal_handlers`

Reply:
don't have a strong opinion.   other programs have a consistent usage?

just want to make sure we respect / don't pollute a namespace elsewhere.

Reply:
done